packages/internals/renderer/src/adapter-store.js

```diff
@@ -0,0 +1,51 @@
+const {remove} = require('lodash');
+const {File} = require('@frctl/support');
+const {toArray} = require('@frctl/utils');
+const debug = require('debug')('fractal:renderer:adapters');
+const Adapter = require('./adapter');
+
+const _adapters = new WeakMap();
+
+class AdapterStore {
+
+  constructor(adapters = []) {
+    _adapters.set(this, []);
+    this.add(adapters);
+  }
+
+  getDefaultAdapter() {
+    return this.adapters[0];
+  }
+
+  getAdapterFor(file) {
+    if (!File.isFile(file)) {
+      throw new Error('Can only retrieve adapters for File objects [file-invalid]');
+    }
+    debug('Finding adapter for file %s: ', file.path);
+    for (const adapter of _adapters.get(this)) {
+      if (adapter.match(file)) {
+        return adapter;
+      }
+    }
+  }
+
+  add(items) {
+    const adapters = _adapters.get(this);
+    toArray(items).map(props => new Adapter(props)).forEach(adapter => {
```

Comment:
I know the Adapter will error here it `props` are invalid, but it might be worth wrapping the error at this level?